cmd/contour/contour.go

```diff
@@ -296,3 +333,32 @@ func getEnv(key, fallback string) string {
 	}
 	return value
 }
+
+// setupTLSConfig sets up a tls.Config, given cert filenames.
+func setupTLSConfig(config *tls.Config, caFile string, servingCert string, servingKey string) error {
+
+	// First up, load the Contour serving cert and key pair
+
+	cert, err := tls.LoadX509KeyPair(servingCert, servingKey)
+	if err != nil {
+		return err
+	}
+
+	ca, err := ioutil.ReadFile(caFile)
+	if err != nil {
+		return err
+	}
+	certPool := x509.NewCertPool()
+
+	if ok := certPool.AppendCertsFromPEM(ca); !ok {
+		return err
```

Comment:
err will be nil there because you checked it on line 347. You'll probably have to do
```
if ...; !ok {
    return fmt.Errorf("Unable to append certificate to pool")
}
```
perhaps there is something in the `ca` value which you can reerence, or not, just use the value of caFile